implementations/rust/ockam/ockam_vault/src/vault/vault_kms/vault_kms.rs

```diff
@@ -140,18 +143,20 @@ impl VaultSecurityModule {
         let attributes = stored_secret.attributes();
         match attributes.secret_type() {
             SecretType::X25519 => {
-                let sk = x25519_dalek::StaticSecret::from(*array_ref![
+                let secret = *array_ref![
                     stored_secret.secret().as_ref(),
                     0,
                     CURVE25519_SECRET_LENGTH_U32 as usize
-                ]);
+                ];
+                let sk = x25519_dalek::StaticSecret::from(secret);
                 let pk = x25519_dalek::PublicKey::from(&sk);
                 Ok(PublicKey::new(pk.to_bytes().to_vec(), SecretType::X25519))
             }
             SecretType::Ed25519 => {
-                let sk = ed25519_dalek::SecretKey::from_bytes(stored_secret.secret().as_ref())
-                    .map_err(|_| VaultError::InvalidEd25519Secret)?;
-                let pk = ed25519_dalek::PublicKey::from(&sk);
+                use ed25519_dalek::SECRET_KEY_LENGTH;
+                let secret = array_ref![stored_secret.secret().as_ref(), 0, SECRET_KEY_LENGTH];
```

Comment:
I think previous version was properly checking the binary length inside the ed25519_dalek library. If we want to cast  it to array ourselves, we need to check the length before doing it, otherwise, it panics. Do you want to do it, or maybe create a community issue for that?

Reply:
We can merge it as is to unblock CI

Reply:
Looks like the `SecretKey` is now forcing you to build the fixed-length array to pass it in the `SigningKey::from_bytes` function. I didn't see another way to do it though.

Let's create an issue for that and merge it as it is.

Reply:
https://github.com/build-trust/ockam/issues/5631